src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -166,18 +170,47 @@ def __init__(
 
     def __getFeatures(self, path):
         """
-        Function to parse features from GeoDataFrame in such a manner that rasterio wants them
+        Function to parse features from GeoDataFrame in such a manner that the NSO api wants them.
 
         @param path: The path to a geojson.
         @return coordinates which rasterio wants to have.
         """
+        gdf = gpd.read_file(path)
         json_loaded = json.loads(gpd.read_file(path).to_json())
+        buffered_polygon = False
 
         if json_loaded["features"][0]["geometry"]["type"] == "MultiPolygon":
-            logging.info("Caution multiploygons are not well supported!")
-            print("Caution multiploygons are not well supported!")
+            logging.info("Multipolygon detected!")
+            print("Multipolygon detected!")
+
+            logging.info("Buffering to grow the different multipolygons together")
+            print(
+                "Buffering to grow the different multipolygons together in one polygon for retrieving links"
+            )
+
+            gdf = gdf.set_crs("EPSG:4326").to_crs("EPSG:28992")
+            buffer_x = 0
+            while gdf.geometry.iloc[0].geom_type != "Polygon":
+                gdf["geometry"] = unary_union(gdf["geometry"].buffer(buffer_x))
+                buffer_x = buffer_x + 1
```

Comment:
Why would you not use a smarter search algorithm here? I.e. a binary search type algorithm. This sounds computationally quite expensive.